linkerd/app/inbound/src/lib.rs

```diff
@@ -274,11 +274,11 @@ impl Config {
         // fails, skips that stack to forward to the local endpoint.
         svc::stack(switch_loopback)
             .check_new_service::<Target, http::Request<http::boxed::BoxBody>>()
-            .push_cache_tracker()
+            .push_cache_tracker(cache_max_idle_age)
             .push_on_response(
                 svc::layers()
                     .push_failfast(dispatch_timeout)
-                    .push_spawn_buffer_with_idle_timeout(buffer_capacity, cache_max_idle_age)
+                    .push_spawn_buffer(buffer_capacity)
```

Comment:
I wonder if removing the idle timeout from buffers means we can just swap this out for `tower-buffer`...might be nice to get rid of the reimplementation. Could be worth doing in a follow-up PR.